docs/builtins.md

```diff
@@ -1742,3 +1742,287 @@ Output
+| ---------- | --------------------------------------------------------------------------------------------------------------------- |
+| `"round"`  | The ends of the line are rounded.                                                                                     |
+| `"butt"`   | The ends of the line are squared off at the endpoints.                                                                |
+| `"square"` | The ends of the line are squared off by adding a box with an equal width and half the height of the line's thickness. |
+
+The initial linecap style is `"round"`.
+
+### `text`
+
+`text` prints text to the canvas at the current cursor position.
+
+#### Example
+
+```evy
+move 20 70
+text "“Time is an illusion."
+move 20 63
+text "Lunchtime doubly so.”"
+move 35 48
+text "― Douglas Adams"
+```
+
+Output
+
+<img width="300" alt="text sample" src="img/text.png">
+
+#### Reference
+
+    text s:string
+
+The `text` function prints the string argument `s` to the canvas at the
+current cursor position. The cursor position is not updated after
+writing text. Only the `fill` and `color` functions update the text
+color, `stroke` has no effect. For more text styling, such as setting
+_font size_ or _font family_, see [`font`](#font).
+
+### `font`
+
+`font` sets font properties for text. The font properties are
+`family`,`size`, `weight`, `style`, `letterspacing`, `baseline`,
+`align`.
+
+#### Example
+
+```evy
+font {family:"Bradley Hand, cursive" size:4}
+
+move 10 65
+text "“The wonderful thing about programming"
+move 10 60
+text "is that anyone can learn it and do it. You"
+move 10 55
+text "don't have to be a genius or have a specific"
+move 10 50
+text "background. You just need curiosity and"
+move 10 45
+text "the willingness to try.”"
+
+// all font properties
+font {
+    size:9
+    style:"normal" // "normal"
+    family:"Tahomana, sans-serif" // see https://developer.mozilla.org/en-US/docs/Web/CSS/font-family
+    weight:900
+    letterspacing:-0.5 // extra inter-character space. negative allowed. default:0
+    align:"right" // "left", "right"
+    baseline:"middle" // "top", "bottom", "alphabetic" (default)
+}
+
+move 90 32
+color "red"
+text "Grace Hopper"
+color "black"
+font {size:4 letterspacing:0 weight:100 style:"normal"}
+move 90 25
+text "computer scientist, compiler builder"
+```
+
+Output
+
+<img width="300" alt="styled text sample" src="img/font.png">
+
+The following example shows the effect of the `align` and `baseline`
+properties:
+
+```evy
+font {size:6 family:"Fira Code, monospace"}
+
+move 25 78
+line 25 86
+move 25 80
+font {align:"left"}
+text "left"
+
+move 25 63
+line 25 71
+move 25 65
+font {align:"right"}
+text "right"
+
+move 25 48
+line 25 56
+move 25 50
+font {align:"center"}
+text "center"
+
+move 55 80
+line 90 80
+move 55 80
+font {baseline:"bottom" align:"left"}
+text "bottom-pq"
```

Comment:
I can't remember why you added `-pq` to this text and I can't figure it out either.

Ah, I remember - we wanted some descenders to show the position relative to the baseline. Hmmm, still seems a little weird but I'm not coming up with an alternative...

Reply:
ok, I've just removed it. It is obvious enough.